homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -159,13 +166,17 @@ async def async_step_prompt_options(self, user_input=None):
             step_id="prompt_options", data_schema=vol.Schema(options), errors=errors
         )
 
-    async def async_step_set_device_options(self, user_input=None):
+    async def async_step_set_device_options(
+        self, user_input: dict[str, Any] | None = None
+    ) -> data_entry_flow.FlowResult:
         """Manage device options."""
         errors = {}
+        devices: dict[str, dict[str, Any] | None] = {}
+        device: dict[str, Any]
+        assert self._selected_device_object
```

Comment:
Given how many times this assertion is needed in this file, I'm thinking it may be worth extracting a helper method to make a property that return the device object?

Reply:
I probably could type it as non nullable instead.

Reply:
That didn't really reduce the asserts. Feels i need to refactor more to get away from them good.